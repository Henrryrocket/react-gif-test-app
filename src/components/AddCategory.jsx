import { useState } from "react"


import PropTypes from "prop-types"

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = (event) => {
        setInputValue(event.target.value);
    }
    const onSubmit = (event ) => {
        event.preventDefault();
        const cleanValue = inputValue.trim();

        if ( cleanValue.length <= 1 ){
            alert('Input a text valid');
            return;
        }
        onNewCategory( cleanValue );
        setInputValue('');
        // setCategories(categories => [inputValue, ...categories])
    }

    
  return (
    <form onSubmit={onSubmit}>
        <input
            type="text"
            placeholder="Search gifs"
            value={ inputValue }
            onChange={ onInputChange }
        />
    </form>
  )
}
AddCategory.protoType = {
    onNewCategory: PropTypes.func.isRequired,
}
