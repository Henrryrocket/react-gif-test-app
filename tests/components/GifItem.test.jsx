import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components";

describe('Tests in <GifItem />', () => {
    const title = 'Test-Title';
    const url = 'https://testpage.com/test.jpg';

    test('should make match with the snapshot', () => { 
        const { container } = render( <GifItem title={ title } url={ url }/> );
        expect(container).toMatchSnapshot();
    });

    test('should show the title and url given', () => { 
        render( <GifItem title={ title } url={ url }/> );
        // screen.debug();
        expect(screen.getByRole('img').src).toBe( url );
    });
});
