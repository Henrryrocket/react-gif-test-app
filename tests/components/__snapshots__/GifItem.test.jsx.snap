// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifItem /> should make match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Test-Title"
      src="https://testpage.com/test.jpg"
    />
    <p>
      Test-Title
    </p>
  </div>
</div>
`;
